src/Compression/Compression.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+using ObjCRuntime;
+
+namespace Compression {
+
+
+	public partial class CompressionStream : IDisposable {
+
+		[StructLayout(LayoutKind.Sequential)]
+		struct CompressionStreamStruct {
+			public IntPtr Destination;
+			public int DestinationLength;
```

Comment:
This is defined as `size_t` in the header, which is 8 bytes on 64-bit platforms, so this should be `nint`.